packages/ui/src/molecules/AggregateRating/AggregateRating.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { forwardRef, Fragment } from 'react'
+import type { FC, HTMLAttributes, ReactNode } from 'react'
+
+import List from '../../atoms/List'
+
+export interface AggregateRatingProps extends HTMLAttributes<HTMLUListElement> {
+  /**
+   * The current value of the rating, based on the quantity of child elements.
+   */
+  value: number
+  /**
+   * ID to find this component in testing tools (e.g.: cypress, testing library, and jest).
+   */
+  testId?: string
+}
+
+export type RatingItemProps = {
+  children?: ReactNode
+  'data-store-aggregate-rating-item'?: 'full' | 'half' | 'empty'
+  'data-testid'?: string
+}
+
+const RatingItem: FC<RatingItemProps> = ({ children, ...otherProps }) => {
+  if (!React.isValidElement(children)) {
+    return <li {...otherProps}>{children}</li>
+  }
+
+  return <li>{React.cloneElement(children, otherProps)}</li>
+}
+
+const AggregateRating = forwardRef<HTMLUListElement, AggregateRatingProps>(
+  function AggregateRating(
+    { children, testId = 'store-aggregate-rating', value, ...otherProps },
+    ref
+  ) {
+    const fillCheck = (itemValue: number) => {
+      if (itemValue >= 1) {
+        return 'full'
+      }
+
+      if (0 < itemValue && itemValue < 1) {
+        return 'half'
+      }
+
+      return 'empty'
+    }
+
+    return (
+      <List
+        data-store-aggregate-rating
+        variant="unordered"
+        ref={ref}
+        data-testid={testId}
+        {...otherProps}
+      >
+        {React.Children.map(children, (child, index: number) => {
+          const currentItemNumber = index + 1
+          const fill = fillCheck(value - currentItemNumber)
```

Comment:
While checking out the component in the preview link, I noticed the [Aggregate Rating](https://deploy-preview-1037--faststoreui.netlify.app/?path=/story/molecules-aggregaterating--aggregate-rating) and [Aggregate Rating With CSS](https://deploy-preview-1037--faststoreui.netlify.app/?path=/story/molecules-aggregaterating--aggregate-rating-with-css) examples have 1 less full-star than the current `value` (e.g. `value={3}` rendered a 2 full stars rating). The [Aggregate Rating Without Icon](https://deploy-preview-1037--faststoreui.netlify.app/?path=/story/molecules-aggregaterating--aggregate-rating-without-icon) example is also rendering `data-store-aggregate-rating-item="empty"`, but showing a full-star.

Aggregate Rating|Aggregate Rating Without Icon
-|-
<img width="1041" alt="Screen Shot 2021-11-29 at 19 22 45" src="https://user-images.githubusercontent.com/381395/143952356-dcd8fad3-1ddb-4c4b-ba51-10659ab70edc.png">|<img width="1037" alt="Screen Shot 2021-11-29 at 19 23 15" src="https://user-images.githubusercontent.com/381395/143952363-5c65f30d-9012-40b9-8421-64e36668753b.png">

Not sure the problem is here, but it's worth investigating.